lib/charms/tls_certificates_interface/v0/tls_certificates.py

```diff
@@ -364,6 +365,29 @@ def __init__(
         self.charm = charm
         self.common_name = common_name
         self.sans = sans
+        self.cert_type = cert_type
+
+        if cert_type is not None and common_name is not None:
+            self.framework.observe(
+                charm.on[relationship_name].relation_joined, self._on_relation_joined
+            )
+
+    def _on_relation_joined(self, event) -> None:
```

Comment:
I', not sure we want this behaviour, I'd rather have the developer ask for the certificate they need when they need it.

Reply:
This PR doesn't change old behaviour. It watches `_on_relation_joined` event only if `cert_type is not None and common_name is not None`

Reply:
The more I think of it, the more I have the opinion that we shouldn't automatically ask for certificates on relation joined. This will abstract the request for certificate and make the charm author think that they don't have to think about certificate. Especially at renewal time. I believe they should explicitely ask for the certificate using the request_certificate method.

Reply:
Yeh, I think I agree with Guilaume here. There are too many assumptions at play if we automatically issue.